File: src/python/oop/ParkingLot/parking/fee/hourly_rate_strategy.py
from .rate_strategy_base import RateStrategyBase
from ..ticket import Ticket

class HourlyRateStrategy(RateStrategyBase):
    __PEAK_HOUR_RATE_MULTIPLIER = 1.5

    def calculate(self, ticket: Ticket, current_rate: float) -> float:
        hour = ticket.park_time.hour
        if (hour >= 7 and hour <= 10) or (hour >= 16 and hour <= 19):
            return current_rate * self.__PEAK_HOUR_RATE_MULTIPLIER

        return current_rate

File: src/go/oop/ParkingLot/internal/fees/hourly_rate_strategy.go
package fees

import (
	"github.com/dobermanch/dev-quests/parking-lot/internal/models"
)

type HourlyRateStrategy struct{}

var (
	peakHourRate float32 = 1.5
)

func (s HourlyRateStrategy) Calculate(ticket models.Ticket, currentRate float32) float32 {
	hour := ticket.ParkingTime.Hour()
	if (hour >= 7 && hour <= 10) || (hour >= 16 && hour <= 19) {
		return currentRate * peakHourRate
	}

	return currentRate
}
